src/index.js

```diff
@@ -438,16 +438,20 @@ function _renderToString(
 			rendered != null && rendered.type === Fragment && rendered.key == null;
 		rendered = isTopLevelFragment ? rendered.props.children : rendered;
 
-		try {
-			// Recurse into children before invoking the after-diff hook
-			const str = _renderToString(
+		const renderChildren = () =>
+			_renderToString(
 				rendered,
 				context,
 				isSvgMode,
 				selectValue,
 				vnode,
 				asyncMode
 			);
+
+		try {
+			// Recurse into children before invoking the after-diff hook
+			const str = renderChildren();
```

Comment:
Worth benchmarking in the future. Creating inline closures is likely slower than inlining it into the various places.